python/grass/__init__.py

```diff
@@ -75,16 +93,48 @@ def _translate(text):
             # does not raise an exception even if the locale settings is broken
             # or the translation files were not installed.
             fallback = True
-            translation = gettext.translation(
-                "grasslibs", locale_dir, fallback=fallback
-            )
-            # Add other domains as fallback.
-            translation.add_fallback(
-                gettext.translation("grassmods", locale_dir, fallback=fallback)
-            )
-            translation.add_fallback(
-                gettext.translation("grasswxpy", locale_dir, fallback=fallback)
-            )
+            translation = None
+            domains = ("grasslibs", "grassmods", "grasswxpy")
+
+            def apply_trans(kwargs):
+                nonlocal translation
+                if kwargs["domain"] == "grasslibs":
+                    translation = gettext.translation(**kwargs)
+                else:
+                    translation.add_fallback(gettext.translation(**kwargs))
+
+            try:
+                # Get user defined lang
+                lang = _UserSettings.Get(
+                    group="language",
+                    key="locale",
+                    subkey="lc_all",
+                )
+            except NameError:
+                # If _UserSettings isn't imported during tests (without wxGUI)
+                lang = None
+
+            if lang and lang not in ("en", "system"):
+                for domain in domains:
+                    apply_trans(
+                        {
+                            "domain": domain,
+                            "localedir": locale_dir,
+                            "fallback": fallback,
+                            "languages": (lang,),
+                        }
+                    )
+            elif lang == "en":
+                translation = gettext.NullTranslations()
```

Comment:
Is this really needed? How that worked before? It is not the case of GRASS GIS, but in a very theoretical level, there could be a translation from developer English to a proper one or to a its dialect.

Reply:
At the moment translations in GUI are identified by string that should match suffix of translation file. Thus if somebody decides to translate GRASS to some flavour of English, he'd use that flavour in a file name e.g. en_GB and thus it would show up in GUI as en_GB (see pt_BR for an example). IIRC "en" option was introduced as Markus had problems using GRASS in Japan :D

Reply:
Yes I understand, but "en" language code represent language if no translation is used. Check my previous comment with use case examples.